feature/select-subject-age-group/src/main/java/com/simprints/feature/selectagegroup/screen/SelectSubjectAgeGroupViewModel.kt

```diff
@@ -0,0 +1,92 @@
+package com.simprints.feature.selectagegroup.screen
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.simprints.core.ExternalScope
+import com.simprints.core.livedata.LiveDataEventWithContent
+import com.simprints.core.livedata.send
+import com.simprints.core.tools.time.TimeHelper
+import com.simprints.core.tools.time.Timestamp
+import com.simprints.feature.exitform.ExitFormConfigurationBuilder
+import com.simprints.feature.exitform.exitFormConfiguration
+import com.simprints.feature.exitform.scannerOptions
+import com.simprints.infra.config.store.ConfigRepository
+import com.simprints.infra.config.store.models.AgeGroup
+import com.simprints.infra.config.store.models.GeneralConfiguration
+import com.simprints.infra.events.SessionEventRepository
+import com.simprints.infra.events.event.domain.models.AgeGroupSelectionEvent
+import com.simprints.infra.logging.LoggingConstants.CrashReportTag.SESSION
+import com.simprints.infra.logging.Simber
+import com.simprints.infra.resources.R
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+internal class SelectSubjectAgeGroupViewModel @Inject constructor(
+    private val timeHelper: TimeHelper,
+    private val eventRepository: SessionEventRepository,
+    private val buildAgeGroups: BuildAgeGroupsDescriptionUseCase,
+    private val configurationRepo: ConfigRepository,
+    @ExternalScope private val externalScope: CoroutineScope,
+) : ViewModel() {
+
+    val finish: LiveData<LiveDataEventWithContent<AgeGroup>>
+        get() = _finish
+    private var _finish = MutableLiveData<LiveDataEventWithContent<AgeGroup>>()
+    val ageGroups: LiveData<List<AgeGroupDisplayModel>>
+        get() = _ageGroupsDisplayModel
+    private var _ageGroupsDisplayModel = MutableLiveData<List<AgeGroupDisplayModel>>()
+    private lateinit var startTime: Timestamp
+
+    val showExitForm: LiveData<LiveDataEventWithContent<ExitFormConfigurationBuilder>>
+        get() = _showExitForm
+    private val _showExitForm =
+        MutableLiveData<LiveDataEventWithContent<ExitFormConfigurationBuilder>>()
+
+    fun start() = viewModelScope.launch {
+        startTime = timeHelper.now()
+        val ageGroups = buildAgeGroups()
+        // notify the adapter
+        _ageGroupsDisplayModel.value = ageGroups
+    }
+
+    fun saveAgeGroupSelection(ageRange: AgeGroup) = externalScope.launch {
```

Comment:
Should you not check that age group is supported by the configuration and if it's not the case display an `AGE_GROUP_NOT_SUPPORTED` alert?

Reply:
Milen is handling the activation of the correct SDK and displaying the alert screen if the age group is not supported. In this PR, my focus is on displaying the UI and saving the selected option events.